go/logic/migrator.go

```diff
@@ -245,6 +253,16 @@ func (this *Migrator) onChangelogStateEvent(dmlEvent *binlog.BinlogDMLEvent) (er
 	return nil
 }
 
+func (this *Migrator) onChangelogHeartbeatEvent(dmlEvent *binlog.BinlogDMLEvent) (err error) {
+	changelogHeartbeatString := dmlEvent.NewColumnValues.StringColumn(3)
+	if lag, err := parseChangelogHeartbeat(changelogHeartbeatString); err != nil {
+		return this.migrationContext.Log.Errore(err)
+	} else {
+		atomic.StoreInt64(&this.migrationContext.CurrentHeartbeatLag, int64(lag))
```

Comment:
I just realised that it would be more correct to store the timestamp here instead of computing and storing the lag.

We should only be computing the lag when we need it (in `printStatus` & `cutOver`) so that it's as fresh as possible. This would be a more correct solution.

It's late here in Dublin, so I will make this improvement tomorrow.

Reply:
Updated the PR